engine/src/main/java/com/sandboni/core/engine/Processor.java

```diff
@@ -107,8 +109,17 @@ private Builder getBuilder() {
     private boolean proceed(ChangeScope<Change> changeScope) {
         return (!isRunAllExternalTests() || !isIntegrationStage())
                 && (arguments.isRunSelectiveMode()
-                || (ChangeScopeAnalyzer.onlySupportedFiles(changeScope, getSupportedFiles())
-                    && ChangeScopeAnalyzer.analyzeConfigurationFiles(changeScope, getBuildFiles())));
+                || (meetsVolatilityRequirements(changeScope)));
+    }
+
+    private boolean meetsVolatilityRequirements(ChangeScope<Change> changeScope){
```

Comment:
Can you derive the scope based on only two flags instead of using this "volatility" value? 
runSelectiveMode (for config changes)
runXXXXMode (find a name) (for non java changes)
and you can set these flags from command line, then the core is "volatility level" agnostic, as I think that this value is only relevant to the context, in this case, the Jules pipeline. Please check if that's possible.
And let's try other words instead of "volatility", is not clear to me the purpose of this value.